lib/statsd.js

```diff
@@ -42,18 +43,37 @@ var Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
       maxBufferSize : maxBufferSize,
       bufferFlushInterval: bufferFlushInterval,
       telegraf    : telegraf,
-      sampleRate    : sampleRate
+      sampleRate  : sampleRate,
+      protocol    : protocol
     };
   }
 
+  function createSocket(args) {
+    var socket;
+
+    if (args.protocol === 'tcp') {
+      socket = net.connect(args.port, args.host);
+      socket.setKeepAlive(true);
+    } else {
+      socket = dgram.createSocket('udp4');
+    }
+
+    return socket;
+  }
+
   // hidden global_tags option for backwards compatibility
   options.globalTags = options.globalTags || options.global_tags;
 
+  this.protocol = options.protocol;
```

Comment:
Nitpick: Although it's against various code styles, I'd stick to the code style already in place and add the alignment whitespaces:

```
- this.protocol = options.protocol;
+ this.protocol    = options.protocol;
```